README.md

```diff
@@ -1,147 +1,181 @@
 
-#### Usage
+**Example**
+```r
+testdata <- '{"X1":[1,2,3],"X2":[4,5,6]}'
+model.predict(data.frame(jsonlite::fromJSON(testdata),stringsAsFactors=TRUE))
 
-`promote.deploy(name,confirm=False)
+```
+<hr>
 
-#### Arguments
-- `name`(_string_): the name of the model to deploy to Alteryx Promote
-- `confirm`(_boolean_, optional): If `TRUE`, then user will be prompted to confirm deployment
+#### `promote.library()`
+Tell the Promote servers to install a library required to run `model.predict()`
 
-#### Examples
+**Arguments**
+- `name`(_string_): the name of the library to install on the Promoter server
+- `src`(_string_, optional): where to look for the library. options:['CRAN' (default), 'github']
+- `version`(_string_, optional): the library version to install (defaults to `NULL` which installs the latest version)
+- `user`(_string_, optional): the github username if `src="github"` (defaults to `NULL`)
+- `install`(_boolean_, optional): whether or not to install the library in the deployed model container (defaults to `TRUE`)
 
-Deploy the "IrisClassifier_model" and don't require confirmation on deployment.
+**Example**
 ```r
-promote.deploy("IrisClassifier_model",confirm=FALSE)
+promote.library("randomforest")
+promote.library("plyr", src='CRAN', version=NULL, user=NULL, install=TRUE)
 ```
 <hr>
 
-### `promote.metadata()`
-
-Store custom metadta about amoel as part of the `model.predict()` when it is sent to the Promote servers. (limited to 6 key-value pairs)
+#### `promote.metadata()`
+Store custom metadata about a model as part of the `model.predict()` when it is sent to the Promote servers. (limited to 6 key-value pairs)
 
-#### Usage
-
-`promote.metadata(name, value)`
-
-#### Arguments
+**Arguments**
 - `name`(_string_): the name of your metadata (limit 20 characters)
 - `value`: a value for your metadata (will be converted to string and limited to 50 characters)
 
-#### Examples
-
+**Example**
 ```r
 promote.metadata("one", 1)
-promote.metadata("two","2")
-promote.metadata("list",list(a=1,b=2))
+promote.metadata("two", "2")
+promote.metadata("list", list(a=1,b=2))
 ```
 <hr>
 
-### `promote.library()`
+#### `promote.config`
+To deploy models, add a username, API key, and URL to the `promote.config` variable
 
-Tell the Promote servers to install a package needed to run `model.predict()`
+**Arguments**
+- `username`(_string_): the username used to sign into the Promote app
+- `apikey`(_string_): the random API key that is assigned to that username
+- `env`(_string_): the URL that can be used to access the Promote app's frontend
 
-#### Usage
+**Example**
+```r
+promote.config <- c(
+  username = "username",
+  apikey = "apikey",
+  env = "http://promote.company.com/"
+)
+```
+<hr>
 
-`promote.library(package)`
+#### `promote.deploy()`
```

Comment:
I think the promote.deploy head (without the parens) is best. I only included the parens to imply that it was a method, since something like promote.config is not a method.